nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestReplaceText.java

```diff
@@ -1074,12 +1090,11 @@ public void testRegexWithBadCaptureGroup() throws IOException {
         runner.setProperty(ReplaceText.REPLACEMENT_STRATEGY, ReplaceText.REGEX_REPLACE);
         runner.setProperty(ReplaceText.EVALUATION_MODE, ReplaceText.ENTIRE_TEXT);
 
+        exception.expect(AssertionError.class);
+        exception.expectMessage("java.lang.IndexOutOfBoundsException: No group 1");
```

Comment:
At first I was really concerned about this change to IndexOutOfBoundsException, because it means the flowfile will rollback and admin yield the ReplaceText.  But if this is simply verifying behavior when the SEARCH_VALUE is incorrectly configured, then I think this is OK.  Can you confirm?  Normally we would prefer a customValidate() method catch this scenario (REPLACEMENT_VALUE references a capture group that does exist in the SEARCH_VALUE) but perhaps that isn't viable at validation time.

Reply:
I'm looking into it, but the processor already has handing for this at runtime not at validation time, where it escapes back references that are over the actual capture count

Reply:
so the issue is this:
```java
 // If we find a back reference that is not valid, then we will treat it as a literal string. For example, if we have 3 capturing
    // groups and the Replacement Value has the value is "I owe $8 to him", then we want to treat the $8 as a literal "$8", rather
    // than attempting to use it as a back reference.
    private static String escapeLiteralBackReferences(final String unescaped, final int numCapturingGroups) {
        if (numCapturingGroups == 0) {
            return unescaped;
        }
```

If there are no capture groups, we don't escape all the findings

Reply:
OK, I made it escape even if there are no named captures, reverted the changed test.